### PR TITLE
Adding Microsoft.Maintenance RP PS cmdlets

### PR DESCRIPTION
**Pull request by abkmr**
## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
@abkmr please fix build errors. THanks

**Comment by isra-fel** [comment]
@abkmr  Hi, any updates on this PR?

**Comment by abkmr** [comment]
> @abkmr Hi, any updates on this PR?

I have updated the tests and those failures are resolved. But I am seeing failures in Help docs that I am unable to solve. Can you help identify why its failing here, and how I can repro it locally? https://dev.azure.com/azure-sdk/public/_build/results?buildId=183343&view=logs&j=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&t=9a0cd9ed-0b8b-525f-138d-85a363b757e6&l=128

**Comment by abkmr** [comment]
@isra-fel The Design review was completed and closed last week, https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/419. Please complete the checkin.

**Comment by abkmr** [comment]
@isra-fel I have updated the PR as per your suggestions. Please see if you can complete the PR

**Comment by abkmr** [comment]
@isra-fel I have updated the PR as per your suggestions. Please see if you can complete the PR. We are near to launch date and need to get it checked in.

File: src/Maintenance/Maintenance/ConfigurationAssignment/ConfigurationAssignmentGetMethod.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
+//-----------------------------------------------------------------------
+// <copyright company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//   Refer to the class documentation.
+// </summary>
+//-----------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (abkmr) [reply src/Maintenance/Maintenance/ConfigurationAssignment/ConfigurationAssignmentGetMethod.cs]: this is there in other ps cmdlet files for other RPs. Added to all files.

File: src/Maintenance/Maintenance/MaintenanceConfiguration/MaintenanceConfigurationDeleteMethod.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Maintenance
+{
+    [Cmdlet(VerbsCommon.Remove, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "MaintenanceConfiguration", DefaultParameterSetName = "DefaultParameter", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSMaintenanceConfiguration))]
```
[the remark this answers is not in the document]
Comment (abkmr) [reply src/Maintenance/Maintenance/MaintenanceConfiguration/MaintenanceConfigurationDeleteMethod.cs]: updated to bool for the return type and response

File: src/Maintenance/Maintenance/MaintenanceConfiguration/MaintenanceConfigurationDeleteMethod.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                        this.ShouldContinue(Properties.Resources.ResourceRemovalConfirmation,
+                                            "Remove-AzMaintenanceConfiguration operation")))
+                {
+                    string resourceGroupName = this.ResourceGroupName;
+                    string name = this.Name;
+
+                    var result = MaintenanceConfigurationsClient.Delete(resourceGroupName, name);
+                    var psObject = new PSMaintenanceConfiguration();
+                    MaintenanceAutomationAutoMapperProfile.Mapper.Map<MaintenanceConfiguration, PSMaintenanceConfiguration>(result, psObject);
+                    WriteObject(psObject);
```
[the remark this answers is not in the document]
Comment (abkmr) [reply src/Maintenance/Maintenance/MaintenanceConfiguration/MaintenanceConfigurationDeleteMethod.cs]: updated to bool for the return type and response with the passthu flag

File: src/Maintenance/Maintenance/MaintenanceConfiguration/MaintenanceConfigurationCreateOrUpdateMethod.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 3,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false)]
+        [AllowNull]
```
[the remark this answers is not in the document]
Comment (abkmr) [reply src/Maintenance/Maintenance/MaintenanceConfiguration/MaintenanceConfigurationCreateOrUpdateMethod.cs]: removed [AllowNull]

File: tools/StaticAnalysis/Exceptions/Az.Maintenance/SignatureIssues.csv [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+"Microsoft.Azure.PowerShell.Cmdlets.Maintenance.dll","Microsoft.Azure.Commands.Maintenance.GetAzureRmConfigurationAssignment","Get-AzConfigurationAssignment","1","8420","Parameter set 'DefaultParameter' of cmdlet 'Get-AzConfigurationAssignment' contains at least one parameter with a position larger than four, which is discouraged.","Limit the number of positional parameters in a single parameter set to four or fewer."
+"Microsoft.Azure.PowerShell.Cmdlets.Maintenance.dll","Microsoft.Azure.Commands.Maintenance.GetAzureRmUpdate","Get-AzMaintenanceUpdate","1","8420","Parameter set 'DefaultParameter' of cmdlet 'Get-AzMaintenanceUpdate' contains at least one parameter with a position larger than four, which is discouraged.","Limit the number of positional parameters in a single parameter set to four or fewer."
```
[the remark this answers is not in the document]
Comment (abkmr) [reply tools/StaticAnalysis/Exceptions/Az.Maintenance/SignatureIssues.csv]: Yes, i have removed the additional positional comments and removed the exceptions from this file

File: src/Maintenance/Maintenance/Az.Maintenance.psd1 [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+DotNetFrameworkVersion = '4.7.2'
+
+# Minimum version of the common language runtime (CLR) required by this module. This prerequisite is valid for the PowerShell Desktop edition only.
+# CLRVersion = ''
+
+# Processor architecture (None, X86, Amd64) required by this module
+# ProcessorArchitecture = ''
+
+# Modules that must be imported into the global environment prior to importing this module
+RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '1.6.2'; })
```
[the remark this answers is not in the document]
Comment (isra-fel) [reply src/Maintenance/Maintenance/Az.Maintenance.psd1]: This is still pending

File: src/Maintenance/Maintenance/help/Az.Maintenance.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+---
+Module Name: Az.Maintenance
+Module Guid: {{ Update Module Guid }}
```
Comment (isra-fel) [inline src/Maintenance/Maintenance/help/Az.Maintenance.md]: Update this with https://github.com/Azure/azure-powershell/pull/10313/files#diff-9b895a4cb333f6bf9176b1eeb9260782R21

File: src/Maintenance/Maintenance/help/Get-AzApplyUpdate.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+Required: True
+Position: 6
+Default value: None
+Accept pipeline input: True (ByPropertyName)
+Accept wildcard characters: False
+```
+
+### -ResourceParentName
+{{ Fill ResourceParentName Description }}
```
Comment (isra-fel) [inline src/Maintenance/Maintenance/help/Get-AzApplyUpdate.md]: Still plenty parameters without help messages. That's why you get these `{{ Fill ... }}`

File: src/Maintenance/Maintenance/ConfigurationAssignment/ConfigurationAssignmentDeleteMethod.cs [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+                    if (!string.IsNullOrEmpty(resourceParentType) && !string.IsNullOrEmpty(resourceParentName))
+                    {
+                        ConfigurationAssignmentsClient.DeleteParent(resourceGroupName, providerName, resourceParentType, resourceParentName, resourceType, resourceName, configurationAssignmentName);
+                    }
+                    else
+                    {
+                        ConfigurationAssignmentsClient.Delete(resourceGroupName, providerName, resourceType, resourceName, configurationAssignmentName);
+                    }
+
+                    if (this.PassThru)
```
Comment (isra-fel) [inline src/Maintenance/Maintenance/ConfigurationAssignment/ConfigurationAssignmentDeleteMethod.cs]: Always use `.IsPresent` to tell if a switch parameters is specified
```suggestion
                    if (PassThru.IsPresent)
```

File: src/Maintenance/Maintenance/Az.Maintenance.psd1 (hunk 6 above)
[the remark this answers is not in the document]
Comment (abkmr) [reply src/Maintenance/Maintenance/Az.Maintenance.psd1]: yeah this got modified back to 1.6.2 due to a local change

File: src/Maintenance/Maintenance/help/Get-AzApplyUpdate.md (hunk 8 above)
Comment (abkmr) [reply src/Maintenance/Maintenance/help/Get-AzApplyUpdate.md]: these are all resolved

File: src/Maintenance/Maintenance/ConfigurationAssignment/ConfigurationAssignmentCreateOrUpdateMethod.cs [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+            });
+        }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
```
[the remark this answers is not in the document]
Comment (abkmr) [reply src/Maintenance/Maintenance/ConfigurationAssignment/ConfigurationAssignmentCreateOrUpdateMethod.cs]: done

**Comment by isra-fel** [comment]
One case (datalake) failed on linux but all tests passed on mac. This PR should be able to merge.

**Pull request merged**